### PR TITLE
Do not merge until Preview 7: dotnet user-jwts /1

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Contributes to #26155
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/jwt-authn?view=aspnetcore-7.0&branch=pr-en-us-26551)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@captainsafia please review. Note, my instructions worked last week. Today when I test I get the following error in the curl console:

```
HTTP/1.1 401 Unauthorized
Content-Length: 0
Date: Wed, 27 Jul 2022 03:09:35 GMT
Server: Kestrel
WWW-Authenticate: Bearer error="invalid_token", error_description="The signature key was not found"
```

And there's no logging information to help out.

File: aspnetcore/security/authentication/jwt-authn.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+### Create
+
+Usage: `dotnet user-jwts create [options]`
+
+| Option  | Description |
+| ------------- | ------------- |
+|  -p \| --project | The path of the project to operate on. Defaults to the project in the current directory. |
+| --scheme | The scheme name to use for the generated token. Defaults to 'Bearer'. |
+| -n \| --name | The name of the user to create the JWT for. Defaults to the current environment user. |
+| --audience | The audiences to create the JWT for. Defaults to the URLs configured in the project's launchSettings.json. |
```
Comment (gfoidl) [inline aspnetcore/security/authentication/jwt-authn.md:59]: Is it really `launchSettings.json`?
In the example below they are given in `appsettings.Development.json`.

File: aspnetcore/security/authentication/jwt-authn.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+## Synopsis
+
+```dotnetcli
+dotnet user-jwts [<PROJECT>|<SOLUTION>] [command]
+dotnet user-jwts [command] -h|--help
+```
+
+## Description
+
+Create and manages app specific local JSON Web Tokens.
```
Comment (gfoidl) [inline aspnetcore/security/authentication/jwt-authn.md:31]: Here I'm missing one sentence on how the "app specific" tokens are created.
With special play-together with the JWT-middleware or how does it work.

It doesn't need a thorough description, just a one-line hint or that like.
As of now it makes me wonder how it works, so it's hard for me to put it into my "toolbox".

Comment (captainsafia) [reply aspnetcore/security/authentication/jwt-authn.md:31]: I think "project-specific" might be a clearer term than app-specific. The user-jwts tool builds on some infrastructure provided by the user secrets manager, which are project specific.

So you can use the same key for the same project (which usually maps to an app) anyways.

File: aspnetcore/security/authentication/jwt-authn.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+## Arguments
+
+`PROJECT | SOLUTION`
+
+The MSBuild project or solution to apply a command on. If a project or solution file is not specified, MSBuild searches the current working directory for a file that has a file extension that ends in *proj* or *sln*, and uses that file.
```
Comment (captainsafia) [inline aspnetcore/security/authentication/jwt-authn.md:37]: It only works for projects, not solutions, at the moment.

File: aspnetcore/security/authentication/jwt-authn.md (hunk 1 above)
Comment (captainsafia) [reply aspnetcore/security/authentication/jwt-authn.md:59]: > Is it really launchSettings.json?

Yep.

> In the example below they are given in appsettings.Development.json.

We parse launch settings for the ports that your app is served on and then add those to `appsettings.Development.json` under the `Authentication:Schemes:Bearer:Audience` category so they can be consumed by the options-from-config infrastructure at runtime.

File: aspnetcore/security/authentication/jwt-authn.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+`~/.microsoft/usersecrets/<secrets_GUID>/secrets.json`
+
+---
+
+### Create a JWT
+
+The following commands create a local JWT:
+
+```dotnetcli
+dotnet user-secrets init
```
Comment (captainsafia) [inline aspnetcore/security/authentication/jwt-authn.md:109]: You no longer need to call `dotnet user-secrets init` before calling the tool.

File: aspnetcore/security/authentication/jwt-authn.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+dotnet user-jwts create
+```
+
+In a future version, `dotnet user-jwts create` will call `dotnet user-secrets init`.
+
+The preceding command creates a JWT and updates the project’s `appsettings.Development.json` file with the JSON similar to the following:
+
+:::code language="csharp" source="~/security/authentication/jwt-authn/samples/MyJWT/appsettings.Development.json" highlight="8-21":::
+
+Copy the JWT and the `ID` created in the preceding command. The following command displays  JWT security information, including expiration, scopes, roles, token header and payload, and the compact token:
```
Comment (captainsafia) [inline aspnetcore/security/authentication/jwt-authn.md:119]: As of preview7, the token gets printed when you create it so you don't have to call the print command to get it out. We probably want to document that flow.

**Pull request merged**